cli/azd/internal/repository/app_init.go

```diff
@@ -908,3 +946,23 @@ func promptSpringBootVersion(console input.Console, ctx context.Context) (string
 		return appdetect.UnknownSpringBootVersion, nil
 	}
 }
+
+func appendJavaEurekaClientEnv(svc project.ServiceConfig, javaEurekaServerService project.ServiceConfig) {
+	if svc.Env == nil {
+		svc.Env = map[string]string{}
+	}
+	svc.Env["eureka.client.register-with-eureka"] = "true"
+	svc.Env["eureka.client.fetch-registry"] = "true"
+	svc.Env["eureka.instance.prefer-ip-address"] = "true"
+	// Use '\${}' to escape parsing ${} placeholder in Bicep, just treat it as an Env of application
+	svc.Env["eureka.client.serviceUrl.defaultZone"] = fmt.Sprintf(
+		"\\${%s_BASE_URL}/eureka", strings.ToUpper(javaEurekaServerService.Name))
```

Comment:
Could you please avoid using the property name directly here? Please refer to current implementation of other dependencies like ServiceBus.

Reply:
It's hard to maintain these properties if they exists here and there.

Reply:
Agree that, I try to integrate it with other environment variables.